public/components/custom_panels/custom_panel_table.tsx

```diff
@@ -133,19 +137,44 @@ export const CustomPanelTable = ({
   };
 
   const onClone = async (newName: string) => {
-    const sourcePanel = selectedCustomPanels[0];
-    const { id, ...newPanel } = { ...sourcePanel, title: sourcePanel.title + ' (copy)' };
+    let sourcePanel = selectedCustomPanels[0];
+    try {
+      if (!isUuid(sourcePanel.id)) {
+        // Observability Panel API returns partial record, so for duplication
+        // we will retrieve the entire record and allow new process to continue.
+        const legacyFetchResult = await coreRefs.http!.get(
+          `${CUSTOM_PANELS_API_PREFIX}/panels/${sourcePanel.id}`
+        );
+        sourcePanel = legacyFetchResult.operationalPanel;
+      }
 
-    dispatch(createPanel(newPanel));
+      const { id, ...newPanel } = {
+        ...sourcePanel,
+        title: newName,
+      };
+
+      dispatch(createPanel(newPanel));
+    } catch (err) {
+      console.log(err);
```

Comment:
set error toast? I do not see toast code here?

Reply:
If toast is not ready, can we use `console.error`?